apps/gi-frontend/src/app/artifacts/Content.tsx

```diff
@@ -21,11 +21,29 @@ export default function Content({
   const [artifacts, setArtifacts] = useState(serverArtifacts)
   const addArtifact = async () => {
     try {
-      const { error } = await supabase.from('artifacts').insert({
-        ...randomizeArtifact(),
-        account: accountId,
-      } as any)
-      if (error) console.error(error)
+      const randArtifact = randomizeArtifact()
+      const { substats, location, ...rest } = randArtifact
+      const { error, data } = await supabase
+        .from('artifacts')
+        .insert({
+          ...rest,
+          location: location || null,
+          account_id: accountId,
+        } as any)
+        .select()
+      if (error) return console.error(error)
+      const art = data[0]
+      if (!art.id) return
+      // TODO: is there a better way to add substats? this sends like 5 requests per artifact...
+      substats.map(async (substat, index) => {
+        if (!substat.key) return
+        const { error } = await supabase.from('substats').insert({
```

Comment:
You can pass an array of objects to `insert`, so you should collect your mapping first and then pass that array